sdks/python/apache_beam/dataframe/convert_test.py

```diff
@@ -85,6 +85,94 @@ def test_convert(self):
       assert_that(pc_3a, equal_to(list(3 * a)), label='Check3a')
       assert_that(pc_ab, equal_to(list(a * b)), label='Checkab')
 
+  def test_convert_memoization(self):
+    with beam.Pipeline() as p:
+      a = pd.Series([1, 2, 3])
+      b = pd.Series([100, 200, 300])
+
+      pc_a = p | 'A' >> beam.Create([a])
+      pc_b = p | 'B' >> beam.Create([b])
+
+      df_a = convert.to_dataframe(pc_a, proxy=a[:0])
+      df_b = convert.to_dataframe(pc_b, proxy=b[:0])
+
+      df_2a = 2 * df_a
+      df_3a = 3 * df_a
+      df_ab = df_a * df_b
+
+      # Converting multiple results at a time can be more efficient.
+      pc_2a_, pc_ab_ = convert.to_pcollection(df_2a, df_ab)
+      # Converting the same expressions should yeild the same pcolls
+      pc_3a, pc_2a, pc_ab = convert.to_pcollection(df_3a, df_2a, df_ab)
+
+      self.assertEqual(id(pc_2a), id(pc_2a_))
+      self.assertEqual(id(pc_ab), id(pc_ab_))
+
+      assert_that(pc_2a, equal_to(list(2 * a)), label='Check2a')
+      assert_that(pc_3a, equal_to(list(3 * a)), label='Check3a')
+      assert_that(pc_ab, equal_to(list(a * b)), label='Checkab')
+
+  def test_convert_memoization_yield_pandas(self):
+    with beam.Pipeline() as p:
+      a = pd.Series([1, 2, 3])
+      b = pd.Series([100, 200, 300])
+
+      pc_a = p | 'A' >> beam.Create([a])
+      pc_b = p | 'B' >> beam.Create([b])
+
+      df_a = convert.to_dataframe(pc_a, proxy=a[:0])
+      df_b = convert.to_dataframe(pc_b, proxy=b[:0])
+
+      df_2a = 2 * df_a
+      df_3a = 3 * df_a
+      df_ab = df_a * df_b
+
+      # Converting multiple results at a time can be more efficient.
+      pc_2a_, pc_ab_ = convert.to_pcollection(df_2a, df_ab,
```

Comment:
Move these tests into the former, so you can also assert that `to_pcollection(x, yield_elements='schema') != to_pcollection(x, yield_elements='pandas')` (i.e. no accidental cross-cache contamination).

Reply:
Done